### PR TITLE
fix: circuitbreaker tests, dependencies and type errors

### PR DESCRIPTION
**Pull request by theartofdev**
Hi @ocampeau ,

I recently came accross your repository during my search for cloud-native examples in Go, which show applications that mitigate endpoint outages due to RPS spikes.
Your utility class looks very useful, and in order to get a better understanding of your implementation, I started studying the test cases. This PR fixes some issues with the tests and type warnings.
By the way, which formatter setup are you using ? I tried to keep the format changes minimal, by switching the VSCode Go plugin from its default to `goformat`.

### CODE REVIEW

File: circuitbreaker/circuit.go [hunk 1]
```diff
@@ -14,8 +15,6 @@ type Options func(breaker *CircuitBreaker)
 var (
   ErrCircuitOpen     = errors.New("http circuit breaker is open")
   ErrCircuitInternal = errors.New("internal error with circuit breaker")
-
-  noopHook = func() {}
```
Comment (theartofdev) [inline circuitbreaker/circuit.go:18]: Seems not being used anywhere ?!

File: circuitbreaker/prommetrics.go [hunk 2]
```diff
@@ -36,7 +37,7 @@ func NewPromCollector(cb *CircuitBreaker) prometheus.Collector {
       nil, prometheus.Labels{LabelsCircuitBreakerName: cb.name}),
   }
 
-  cb.RegisterOnHalfOpenHooks(col.circuitBreakerOpen)
```
Comment (ocampeau) [inline circuitbreaker/prommetrics.go:39]: Nice catch!

File: circuitbreaker/example/plain.go [hunk 3]
```diff
@@ -1,39 +1,43 @@
...
 // in a closure, like here
-func operationInClosure() func() error  {
-  var res = 0
+func operationInClosure() func() error {
   var err error = nil
 
   op := func() error {
-    res, err = someComplexOperation(1, 2, 3)
-    return nil
+    _, err = someComplexOperation(1, 2, 3)
```
Comment (ocampeau) [inline circuitbreaker/example/plain.go:29]: I agree

File: circuitbreaker/circuit.go (hunk 1 above)
Comment (ocampeau) [reply circuitbreaker/circuit.go:18]: Thanks a lot for contributing to this project. I wrote this implementation as a first draft, but it needs more refining and cleanup a bit. I really appreciate the time you took on this. I have not worked on this in a while for a lack of time. but I will in the near future. Please do not hesitate to contribute again if you have anything you want to show.

As for the linter, normally I use standard go tools (go fmt and go vet), but I forgot to setup my pre-commit hook last time I committed.

**Pull request merged**